### ISSUE TITLE
[Safari] Having a gallery with multiple base64-encoded images dramatically increases input lag

### ISSUE THREAD

**Issue by NimaBoscarino** [gradio-app/gradio#1590]
### Describe the bug

As far as I can see, this only happens on Safari, so Chrome and Firefox are not affected. As in the linked notebook, adding image URLs to a Gallery (which are thus being base64 encoded through the post-processing) adds a ton of lag to text inputs.

I looked into the profiler and saw that for some reason calls to the `Textbox.svelte` `resize()` function exploded per keystroke in the textbox after the Gallery is populated, and the number of repeated calls scales with the number of base64 images in the gallery (and also potentially with the resolution of the images).

After a ton of digging, I found that one of the culprits is the `h-full` class added to the `<img>` [in the Gallery component](https://github.com/gradio-app/gradio/blob/6413b65328baaa8a227e0608ccc86b23438c8584/ui/packages/app/src/components/Gallery/Gallery.svelte#L158). Even replacing that class with a `style="height: 100%"` causes the issue 🤷 

This seems like a super funky Webkit issue? I checked the preview version of Safari 16 and the bug is present there too. An easy fix is just to remove the `h-full` class from that component (I don't think that class is necessary for images within a button, but I'm not sure), but I don't know if this bug also affects other components right now. Strangely, the Image component doesn't cause any lag, even though it has `h-full`. So maybe there's something else going on as well?

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Reproduction

https://colab.research.google.com/drive/14BwIP_vgBLKS6Yed4WynyxUkF11lXXHt?usp=sharing

### Logs

```shell
N/A
```

### System Info

```shell
Gradio 3.0.18, macOS 12.5 + Google Colab, with Safari Version 15.5 (17613.2.7.1.8) and Safari Technology Preview Release 146 (Safari 15.4, WebKit 17614.1.14.10.6)
```

### Severity

annoying

**Comment by aliabid94** [comment]
Will take a look, thanks for raising this!

**Comment by pngwn** [comment]
Looks like it is a case of safari being safari. Should be simple enough to workaround but, as you mentioned @NimaBoscarino, we just need to make sure it doesn't impact other components.

Note: We need to make this easier. After 3.1, or during if I get time, I really need to get some decent regression testing cross component testing in place so we can more easily reason about changes like this. Likewise encapsulating the CSS within the component is becoming something of a priority (although it is important for theming also).

**Comment by sashavor** [comment]
Happened to me too! Typing is super slow..

**Comment by evanjs** [comment]
Looks like the surrounding block has changed a bit since the original post
New location: https://github.com/gradio-app/gradio/blob/a05f32c8f1301758dd6957d378253151a52c047b/ui/packages/app/src/components/Gallery/Gallery.svelte#L187
As previously mentioned, simply removing `h-full` from the Image component classes seems to remedy the issues mentioned by https://github.com/AUTOMATIC1111/stable-diffusion-webui/issues/232

Would love to see this change make it upstream eventually, but I have no idea what is required for testing.
As others have mentioned, we want to make sure this doesn't negatively impact other components, etc.

**Comment by pngwn** [comment]
I think we can put a really short debounce on the resize to fix this without needing to touch the layout. It's pretty weird problem but a debounce might be the safest fix.

**Comment by evanjs** [comment]
It seems like #2265 does indeed resolve the issue, at least based on my testing
Thanks guys!

**Comment by pngwn** [comment]
Yep, I can also confirm this issue is no longer present. Thanks for testing @evanjs!
